### PR TITLE
Add support for Top Searches on @faststore/api

### PR DESCRIPTION
**Pull request by victorhmp**
## What's the purpose of this pull request?

This adds support for querying for the most popular search terms in a certain account, powered by VTEX Intelligent Search.

## How it works? 

Now the resolver for the `search.suggestions` field will return the most popular search terms when there was no `term` argument provided in the `search` query sent by the client. It does so by making a GET request to the `intelligent-search` API at `/_v/api/intelligent-search/top_searches` and returning its data.

## How to test it?

Run the package locally using `yarn develop`, and perform the query below.

```graphql
query TopSearchesQuery {
  search(first: 10, term: "") {
    suggestions {
      terms
      products {
	name
      }
    }
  }
}
```

### CODE REVIEW

File: packages/api/src/platforms/vtex/resolvers/query.ts [hunk 1]
```diff
@@ -172,4 +172,16 @@ export const Query = {
...
     )
   },
+  topSearches: async (_: unknown, __: unknown, ctx: Context) => {
+    const {
+      clients: { search },
+    } = ctx
+
+    const { searches } = await search.topSearches()
+
+    return searches.map((searchTermInfo) => ({
```
Comment (igorbrasileiro) [inline packages/api/src/platforms/vtex/resolvers/query.ts:182]: This result is ordered? wdyt about adding a way to order this result?

**Review by igorbrasileiro — APPROVED** [review]
I want to know from where this StoreTopSearchTerm type comes?

**Comment by victorhmp** [comment]
> I want to know from where this StoreTopSearchTerm type comes?

It's almost the exact same as what the Intelligent Search API returns. I didn't want to make something too different from it.

File: packages/api/src/platforms/vtex/resolvers/query.ts (hunk 1 above)
Comment (victorhmp) [reply packages/api/src/platforms/vtex/resolvers/query.ts:182]: I have to confirm this, but I'm pretty sure the API returns the results ordered from most to least searched.

Comment (victorhmp) [reply packages/api/src/platforms/vtex/resolvers/query.ts:182]: About adding a way to control the order... we could add it, but first let me check if the API supports it

Comment (victorhmp) [reply packages/api/src/platforms/vtex/resolvers/query.ts:182]: Just checked with the IS team, this API doesn't support any configuration for now, so I guess it's fine for now to not add any arguments to this query.

**Comment by victorhmp** [comment]
@tlgimenes That's the thing hahaha we thought that it would be confusing for users, since we're efectivelly making a totally different request, and the top searches are not quite "suggestions"

**Comment by tlgimenes** [comment]
it's weird to have a topLevel parameter called topSearches, don't you think? seems way to attached to a specific feature from a specific provider. What about creating an `autocomplete` edge or something like this instead of topSearches?

File: packages/api/src/platforms/vtex/clients/search/index.ts [hunk 2]
```diff
@@ -97,6 +98,9 @@ export const IntelligentSearch = (
   ): Promise<SuggestedTerms> =>
     fetchAPI(`${base}/api/split/suggestion_search/?q=${args.query}`)
 
+  const topSearches = (): Promise<{ searches: Search[] }> =>
```
Comment (emersonlaurentino) [inline packages/api/src/platforms/vtex/clients/search/index.ts:101]: You need to rebase your branch with the main, in order to get the new type for that
```suggestion
  const topSearches = (): Promise<Suggestion> =>
```

File: packages/api/mocks/TopSearchesQuery.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export const TopSearchesQuery = `query TopSearchesQuery {
```
Comment (emersonlaurentino) [inline packages/api/mocks/TopSearchesQuery.ts:1]: Remember to remove this

Comment (victorhmp) [reply packages/api/mocks/TopSearchesQuery.ts:1]: Why? 🤔 This is the mocked requests for the tests to run

**Review by icazevedo — APPROVED** [review]
LGTM

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**